pandas/core/tools/datetimes.py

```diff
@@ -400,6 +403,16 @@ def _convert_listlike_datetimes(
             arg = arg.tz_convert(None).tz_localize("utc")
         return arg
 
+    elif isinstance(arg_dtype, ArrowDtype) and arg_dtype.kind == "M":
+        # TODO: Combine with above if DTI/DTA supports Arrow timestamps
+        if utc:
+            import pyarrow as pa
+
+            # array attribute has to exist since arg is Index/Series at this point
+            arg_arr = arg.array._pa_array
+            arg._pa_array = arg_arr.cast(pa.timestamp(arg_arr.type.unit, "utc"))
```

Comment:
setting _pa_array looks really dangerous  (and weird; can't arg be a Series at this point)?  can we use a astype instead?

Reply:
Thanks, updated.